SPECS/elfutils/elfutils.spec

```diff
@@ -15,15 +15,14 @@ Source0:        https://sourceware.org/elfutils/ftp/%{version}/%{name}-%{version
 BuildRequires:  bison >= 1.875
 BuildRequires:  bzip2-devel
 BuildRequires:  flex >= 2.5.4a
-BuildRequires:  gcc >= 4.1.2-33
+BuildRequires:  gcc
 BuildRequires:  gettext
-BuildRequires:  glibc >= 2.7
 BuildRequires:  m4
+AutoReq:        no
```

Comment:
Disabling `AutoReq` is risky because this forces us to figure out all of `elfutils`'s run-time dependencies manually and keep them up-to-date in the spec. In case of any reshuffling of files in other packages this one depends on, or an addition of a new dependency due to a version change here, we may miss it and the package may not work correctly.

If `elfutils` are currently pulling in the `-devel` packages, then it's possible the specs, which generate these `-devel` packages may be putting run-time libraries into them. In other words - this looks more like a misconfiguration in other packages.

Also, I see that `elfutils` only directly depends on two `-devel` packages: `bzip2-devel` and `elfutils-devel`.
![image](https://github.com/microsoft/azurelinux/assets/24633722/dfcc5e2e-1135-4d70-9942-9d1f1c50de80)

All other ones are probably run-time dependencies of these dependencies, so it's even more possible we're going to lose valid run-time dependencies here by adding `AutoReq: no`.

Reply:
The automatic dependencies added are to shared object files provided by devel packages, for example:
```
$ sudo rpm -q --requires ../build/toolchain/built_rpms_all/elfutils-0.189-1.azl3.x86_64.rpm^C
liblzma.so.5()(64bit)
liblzma.so.5(XZ_5.0)(64bit)
libz.so.1()(64bit)
libz.so.1(ZLIB_1.2.2.3)(64bit)
libzstd.so.1()(64bit)
```
Which pulls in zstd-devel, xz-devel, etc.
I'll see if there is a better way to remove these dependencies. It doesn't look like the Fedora eflutils requires them: https://rpmfind.net/linux/RPM/fedora/updates/testing/40/x86_64/Packages/e/elfutils-0.191-2.fc40.x86_64.html

Reply:
Thanks for the help, the better fix for this is in a fresh PR: https://github.com/microsoft/azurelinux/pull/8373